src/app/login-modal/login-modal.component.ts

```diff
@@ -60,27 +70,58 @@ export class LoginModalComponent implements OnInit {
       return;
     }
 
-    this.loggingIn = true;
+    if (this.loggingIn) {
+      const credentials: Credentials = {
+        captcha_code: captchaResponse,
+        email_address: this.loginForm.value.emailAddress,
+        raw_password: this.loginForm.value.password,
+      };
+  
+      this.identityService.login(credentials).subscribe((response: { id: string, jwt: string }) => {
+        this.identityService.saveJWT(response.id, response.jwt);
+        this.dialogRef.close(response);
+        this.loggingIn = false;
+      }, (error) => {
+        this.captcha.reset();
+        this.loginError = (error.error.description !== undefined ? error.error.description : error.message) || 'Unknown error';
+        this.loggingIn = false;
+      });
+    }
 
-    const credentials: Credentials = {
-      captcha_code: captchaResponse,
-      email_address: this.form.value.emailAddress,
-      raw_password: this.form.value.password,
-    };
+    else if (this.userAskedForResetLink) {
+      this.identityService.getResetPasswordToken(this.resetPasswordForm.value.emailAddress).subscribe((response) => {
+        // console.log(response);
+        // console.log(response.id);
+        // console.log(response.jwt);
+        // this.identityService.saveJWT(response.id, response.jwt);
+        // this.dialogRef.close(response);
+        // this.userAskedForResetLink = false;
+      }, (error) => {
+        // Do NOT surface any error for security reasons. Let the user see the message to check their email for the
+        // password reset link, even if the email entered was invalid. This helps ensure we do not surface any
+        // information.
 
-    this.identityService.login(credentials).subscribe((response: { id: string, jwt: string }) => {
-      this.identityService.saveJWT(response.id, response.jwt);
-      this.dialogRef.close(response);
-      this.loggingIn = false;
-    }, (error) => {
-      this.captcha.reset();
-      this.loginError = (error.error.description !== undefined ? error.error.description : error.message) || 'Unknown error';
-      this.loggingIn = false;
-    });
+        // this.captcha.reset();
+        // this.resetPasswordError = (error.error.description !== undefined ? error.error.description : error.message) || 'Unknown error';
+        // console.log(error);
+        // this.userAskedForResetLink = false;
+      });
+    }
   }
 
   login() {
+    this.loggingIn = true;
     this.captcha.reset();
     this.captcha.execute();
   }
+
+  sendPasswordResetLink() {
+    this.userAskedForResetLink = true;
+    this.captcha.reset();
+    this.captcha.execute();
+  }
+
+  forgotPasswordClicked() {
```

Comment:
```suggestion
  sendPasswordResetLink(): void {
    this.userAskedForResetLink = true;
    this.captcha.reset();
    this.captcha.execute();
  }

  forgotPasswordClicked(): void {
```

Feel free to ignore if you think that's too obvious to be worth stating. Might just be a habit from PHP.

Reply:
Seems a bit inconsistent that one of these is written as an imperative command "sendPasswordResetLink" and one is written as a past tense event "forgotPasswordClicked". Maybe better to make them both commands for consistency with "login", so change "forgotPasswordClicked" to "showPasswordReset" or something?

Reply:
I've renamed `sendPasswordResetLink` to `resetPasswordClicked`